llvm/include/llvm/SYCLLowerIR/PropagateAspectsUsage.h

```diff
@@ -0,0 +1,29 @@
+//===---- PropagateAspectsUsage.h - PropagateAspectsUsage Pass ------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Pass propagates metadata corresponding to usage of optional device
+// features.
+//
+//===----------------------------------------------------------------------===//
+//
+#ifndef LLVM_SYCLPROPAGATE_ASPECTS_USAGE_H
+#define LLVM_SYCLPROPAGATE_ASPECTS_USAGE_H
+
+#include "llvm/IR/PassManager.h"
+
+namespace llvm {
+
+class PropagateAspectsUsagePass
+    : public PassInfoMixin<PropagateAspectsUsagePass> {
+public:
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &);
+};
+
```

Comment:
Are we not creating 'legacy' versions anymore?

Reply:
This pass is supposed to be used from `clang/lib/CodeGen/BackendUtil.cpp`, i.e. from Clang middle-end passes, which use new pass manager without possibility to switch to legacy one